x-pack/plugins/elastic_assistant/server/lib/langchain/elasticsearch_store/elasticsearch_store.ts

```diff
@@ -139,6 +144,26 @@ export class ElasticsearchStore extends VectorStore {
     }
   };
 
+  addDocumentsViaDataClient = async (
+    documents: Document[],
+    options?: Record<string, never>
+  ): Promise<string[]> => {
+    if (!this.kbDataClient) {
+      this.logger.error('No kbDataClient provided');
+      return [];
+    }
+
+    try {
+      const response = await this.kbDataClient.addKnowledgeBaseDocuments({
+        documents,
+      });
+      return response.map((doc) => doc.id);
+    } catch (e) {
+      this.logger.error(`Error loading data into KB\n ${e}`);
+      return [];
```

Comment:
Going to leave this one as-is for now since it matches the current behavior of the `addDocuments()` function. But will revisit this as I work through the tool creation next week and (possibly) move document creation out of esStore.